awswrangler/s3/_merge_upsert_table.py

```diff
@@ -109,10 +121,18 @@ def merge_upsert_table(
     """
     # Check if table exists first
     if wr.catalog.does_table_exist(database=database, table=table, boto3_session=boto3_session):
-        # Read the existing table into a pandas dataframe
-        existing_df = wr.s3.read_parquet_table(database=database, table=table, boto3_session=boto3_session)
+
+        try:
+            # Read the existing table into a pandas dataframe
+            existing_df = wr.s3.read_parquet_table(database=database, table=table, boto3_session=boto3_session)
+
+        except NoFilesFound:
+            # Generate empty frame with the schema specified in the Glue catalog
+            existing_df = _generate_empty_frame_for_table(database=database, table=table, boto3_session=boto3_session)
```

Comment:
The reason I went for generating an empty DataFrame is to be able to take advantage of the data quality tests, including the schema checks. If the table definition in the Glue catalog is incompatible with `delta_df`, the function should throw an error.